test/Microsoft.ML.Tests/Scenarios/SentimentPredictionTests.cs

```diff
@@ -32,8 +32,8 @@ public void TrainAndPredictSentimentModelTest()
                 OutputTokens = true,
                 StopWordsRemover = new PredefinedStopWordsRemover(),
                 VectorNormalizer = TextTransformTextNormKind.L2,
-                CharFeatureExtractor = new NGramNgramExtractor() { NgramLength = 2, AllLengths = true },
-                WordFeatureExtractor = new NGramNgramExtractor() { NgramLength = 3, AllLengths = false }
+                CharFeatureExtractor = new NGramNgramExtractor() { NgramLength = 3, AllLengths = false },
```

Comment:
The purpose of these tests is not to train a great model; if that was a purpose we'd be training on a dataset with more than 250 examples. :) The purpose of these tests is so that we (1) give the code a workout and (2) are alerted when we've changed the behavior of something, which may have been a change we meant to introduce. For that reason, it is important that we test *both* `AllLengths = true` and `AllLengths = false`. If we make them both the same we're reducing code coverage of the test.

Of course, for that same reason, I'd say that issue #25 was an issue in the first place, since the purpose of these tests is to work out the code, not to create a good model.

Our samples are a different matter, since those are meant to express what we might consider to be best practices. But again that's not the point of our tests at all.

---
In reply to: [187114129](https://github.com/dotnet/machinelearning/pull/94#discussion_r187114129) [](ancestors = 187114129)

Reply:
I issue #25 is important as it provides users good defaults. These defaults are benchmarked to work well on many text datasets. 
 
As said, this a toy example, so the actual output metrics aren't the goal. Hence it's ok that they went down. 

The updated code does exercise both `AllLengths=true` and `AllLengths=false`:
```cs
 CharFeatureExtractor = new NGramNgramExtractor() { NgramLength = 3, AllLengths = false },
  WordFeatureExtractor = new NGramNgramExtractor() { NgramLength = 2, AllLengths = true }
```

Reply:
ok then, I'l update my vote.

---
In reply to: [187211456](https://github.com/dotnet/machinelearning/pull/94#discussion_r187211456) [](ancestors = 187211456)